src/library/psi_library.jl

```diff
@@ -1149,3 +1149,415 @@ function build_two_zone_5_bus(; kwargs...)
     end
     return sys
 end
+
+const COST_PERTURBATION_NOISE = rand(1_000_000)
+
+function _duplicate_system(main_sys::PSY.System, twin_sys::PSY.System, HVDC_line::Bool)
+    names = [
+        "114_SYNC_COND_1",
+        "314_SYNC_COND_1",
+        "313_STORAGE_1",
+        "214_SYNC_COND_1",
+        "212_CSP_1",
+    ]
+    for sys in [main_sys, twin_sys]
+        for d in get_components(
+            x -> get_fuel(x) == ThermalFuels.DISTILLATE_FUEL_OIL,
+            ThermalStandard,
+            sys,
+        )
+            for s in get_services(d)
+                remove_service!(d, s)
+            end
+            remove_component!(sys, d)
+        end
+        for d in PSY.get_components(x -> x.name ∈ names, PSY.Generator, sys)
+            for s in get_services(d)
+                remove_service!(d, s)
+            end
+            remove_component!(sys, d)
+        end
+        for d in
+            get_components(x -> get_fuel(x) == ThermalFuels.NUCLEAR, ThermalStandard, sys)
+            set_must_run!(d, true)
+        end
+    end
+
+    # clear time series
+    PSY.clear_time_series!(twin_sys)
+
+    # change names of the systems
+    PSY.set_name!(main_sys, "main")
+    PSY.set_name!(twin_sys, "twin")
+
+    # change the names of the areas and loadzones first
+    for type_ in [PSY.Area, PSY.LoadZone]
```

Comment:
Why `type_` and `name_` instead of `type` and `name`?

Reply:
type isn’t a Julia keyword, right?

Reply:
well it's part of `abstract type`…

Reply:
`The following two-word sequences are reserved: abstract type, mutable struct, primitive type. However, you can create variables with names: abstract, mutable, primitive and type.`
https://docs.julialang.org/en/v1/base/base/#Keywords

Reply:
I just switched to `component_type` for clarity